app/src/main/java/org/mozilla/fenix/library/history/viewholders/HistoryGroupViewHolder.kt

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.history.viewholders
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import org.mozilla.fenix.R
+import org.mozilla.fenix.databinding.HistoryListGroupBinding
+import org.mozilla.fenix.ext.hideAndDisable
+import org.mozilla.fenix.ext.showAndEnable
+import org.mozilla.fenix.library.history.History
+import org.mozilla.fenix.library.history.HistoryFragmentState
+import org.mozilla.fenix.library.history.HistoryInteractor
+import org.mozilla.fenix.library.history.HistoryViewItem
+import org.mozilla.fenix.selection.SelectionHolder
+
+/**
+ * A view representing a search group (a group of history items) in the history list.
+ * [org.mozilla.fenix.library.history.HistoryAdapter] is responsible for creating
+ * and populating the view.
+ */
+class HistoryGroupViewHolder(
+    view: View,
+    private val historyInteractor: HistoryInteractor,
+    private val selectionHolder: SelectionHolder<History>,
+    private val onDeleteClicked: (Int) -> Unit
```

Comment:
We need to pass the removal from the adapter first, sadly.
When the last item in a `timegroup` is being removed, we also need to remove the header. For that, we need to calculate if it's the only item in the group. But we don't have direct access to the items because `PagingData` implementation. The `getItem()` inside the adapter is the only place we could do that.
So, adapter will calculate if additional headers should be filtered out as well, and then passes that data into interactor.